paths.tex

```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
 
 \subsection{Consecutive Shortest Path Distance Algorithm}
 
-\begin{figure}[H]
-\begin{lstlisting}[frame=single]
-  Input : Graph (V,E,T), dist[][][] entries set to Inf
-  Output : dist[][][]
-
-  for (t,src) in {T(e) | e in E } x V:
-    st_e := (src,src)
-    T(e) := (t,t) # or (t, Inf)?
-    R := {st_e}
-    S := {}
-    dist[t][src][src] = 0
-
-    while R != {}:
-      pred_e := pop R
-      for e in edge_neighborhood(pred_e) - union S R:
-        t_int := intersect T(pred_e) T(e)
-        if (t_int != null):
-          dist[t][src][snd e] := min(dist[t][src][fst e] + 1,
-                                        dist[t][src][snd e])
-          insert(e,R)
-      S += pred_e
-    return dist
-\end{lstlisting}
-\caption{An algorithm to compute the lengths of all consecutive temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
```

Comment:
Ok -- I was treating the subscripts as indices totes to resolve types.
On Tue, Mar 22, 2016 at 9:47 AM Eric Campbell <notifications@github.com>
wrote:

> In paths.tex
> <https://github.com/pomonadb/graphstreams/pull/6#discussion_r57022686>:
>
> > -        if (t_int != null):
> > -          dist[t][src][snd e] := min(dist[t][src][fst e] + 1,
> > -                                        dist[t][src][snd e])
> > -          insert(e,R)
> > -      S += pred_e
> > -    return dist
> > -\end{lstlisting}
> > -\caption{An algorithm to compute the lengths of all consecutive temporal shortest
> > -paths}
> > +\begin{algorithm}[H]
> > +\SetAlgoLined
> > +\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
> > +  a complete mapping between pairs of vertices along with an initial timestamp
> > +  to the distance between them, with the default values set to $\infty$.}
> > +\KwOut{The algorithm outputs the 3D array $dist$ such that
> > +  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
>
> Stay consistent with src vs v_{src}. Same for dest and v_{dest}.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pomonadb/graphstreams/pull/6/files/67cd673357118184d4cadc98a30e30a36b8708cc#r57022686>
>